src/tls/boring.rs

```diff
@@ -132,27 +138,23 @@ impl Certs {
         conn.check_private_key()?;
 
         // Ensure that client certificates are validated when present.
-        conn.set_verify(ssl::SslVerifyMode::FAIL_IF_NO_PEER_CERT);
-        conn.set_verify_callback(ssl::SslVerifyMode::PEER, |_, _| {
+        conn.set_verify_callback(Self::verify_mode(), |_, _| {
             // TODO: this MUST verify before upstreaming
             true
         });
         conn.set_alpn_protos(Alpn::H2.encode())?;
 
         Ok(conn.build())
     }
-    pub fn connector(&self) -> Result<ssl::SslConnector, Error> {
+    pub fn connector(&self, id: Identity) -> Result<ssl::SslConnector, Error> {
         let mut conn = ssl::SslConnector::builder(ssl::SslMethod::tls_client())?;
 
         conn.set_private_key(&self.key)?;
         conn.set_certificate(&self.cert)?;
+        conn.set_ca_file("./var/run/secrets/istio/root-cert.pem")?;
```

Comment:
This is the CA of XDS, not workloads; we can do this for short term but we need to fetch it from the returned CSR like we do in Envoy mode long term

Reply:
I've switched it to add the certs from the chain. 

Envoy does this differently.. seems like in their verification callback they check the intermediates.

https://github.com/envoyproxy/envoy/blob/8618789f14fc8314db3063d7ccde265c5198b2bc/source/extensions/transport_sockets/tls/context_impl.cc#L1235

I tried something similar and it failed with `unable to get local issuer certificate`.

At one point I was trying to register the root/intermediates (we ofc just have a root) using `add_extra_chain_cert` but I don't think that does what I think it does. 

https://github.com/envoyproxy/envoy/blob/8618789f14fc8314db3063d7ccde265c5198b2bc/source/extensions/transport_sockets/tls/context_impl.cc#L1285